pl_bolts/datamodules/cifar10_datamodule.py

```diff
@@ -58,125 +54,75 @@ class CIFAR10DataModule(LightningDataModule):
         dm.val_transforms  = ...
     """
 
-    name = 'cifar10'
-    extra_args = {}
+    name = "cifar10"
+    DATASET_CLS = CIFAR10
+    DIMS = (3, 32, 32)
 
     def __init__(
-            self,
-            data_dir: Optional[str] = None,
-            val_split: int = 5000,
-            num_workers: int = 16,
-            batch_size: int = 32,
-            seed: int = 42,
-            *args,
-            **kwargs,
+        self,
+        data_dir: Optional[str] = None,
+        val_split: Union[int, float] = 0.2,
+        num_workers: int = 16,
+        normalize: bool = False,
+        seed: int = 42,
+        batch_size: int = 32,
+        *args,
+        **kwargs,
     ):
         """
         Args:
-            data_dir: where to save/load the data
-            val_split: how many of the training images to use for the validation split
-            num_workers: how many workers to use for loading data
-            batch_size: number of examples per training/eval step
+            data_dir: Where to save/load the data
+            val_split: Percent (float) or number (int) of samples to use for the validation split
+            num_workers: How many workers to use for loading data
+            normalize: If true applies image normalize
+            seed: Seed to fix the validation split
+            batch_size: How many samples per batch to load
         """
-        super().__init__(*args, **kwargs)
 
-        if not _TORCHVISION_AVAILABLE:  # pragma: no cover
-            raise ModuleNotFoundError(
-                'You want to use CIFAR10 dataset loaded from `torchvision` which is not installed yet.'
+        if not _TORCHVISION_AVAILABLE:
+            raise ModuleNotFoundError(  # pragma: no-cover
+                "You want to use CIFAR10 dataset loaded from `torchvision` which is not installed yet."
             )
 
-        self.dims = (3, 32, 32)
-        self.DATASET = CIFAR10
-        self.val_split = val_split
-        self.num_workers = num_workers
-        self.batch_size = batch_size
-        self.seed = seed
-        self.data_dir = data_dir if data_dir is not None else os.getcwd()
-        self.num_samples = 50000 - val_split
+        super().__init__(
+            data_dir=data_dir,
+            val_split=val_split,
+            num_workers=num_workers,
+            normalize=normalize,
+            seed=seed,
+            batch_size=batch_size,
+            *args,
+            **kwargs,
+        )
 
     @property
-    def num_classes(self):
+    def num_samples(self) -> int:
+        len_dataset = 50_000
+        # todo: clean up with parent methods
```

Comment:
What do you mean by `# todo: clean up with parent methods`?

Reply:
this seems to be very similar to the one in `VisionDataModule` but has not checked details